# HG changeset patch
# User Vanya Sergeev <vsergeev@gmail.com>
# Date 1316769181 14400
#      Fri Sep 23 05:13:01 2011 -0400
# Node ID d4404589498c1ef717f585bd8029b0a33a3b4424
# Parent  5b1ddcf5ede1593a123f142ea7d1c6294aa472de
pascal.m: permutation compatibility with Matlab for n=2; fixed n=0 case (Bug #34365)

* pascal.m: permutation compatibility with Matlab for n=2; fixed n=0 case (Bug #34365)

diff --git a/scripts/special-matrix/pascal.m b/scripts/special-matrix/pascal.m
--- a/scripts/special-matrix/pascal.m
+++ b/scripts/special-matrix/pascal.m
@@ -50,43 +50,43 @@ function retval = pascal (n, t)
     error ("pascal: expecting scalar arguments, found something else");
   endif
 
   if (t < -1 || t > 2)
     error ("pascal: expecting T to be -1, 0, 1, or 2, found %d", t);
   endif
 
   retval = zeros (n);
-  retval(:,1) = 1;
+  if (n > 0)
+    retval(:,1) = 1;
+  endif
 
   if (t == -1)
     for j = 2:n
       retval(j:n,j) = cumsum (retval (j-1:n-1,j-1));
     endfor
   else
     for j = 2:n
       retval(j:n,j) = -cumsum (retval (j-1:n-1,j-1));
     endfor
   endif
 
   if (t == 0)
     retval = retval*retval';
   elseif (t == 2)
     retval = retval';
-    retval = retval(n:-1:1,:);
-    retval(:,n) = -retval(:,n);
-    retval(n,:) = -retval(n,:);
+    retval = retval (:,n:-1:1);
     if (rem(n,2) != 1)
       retval = -retval;
     endif
   endif
 
 endfunction
 
 %!assert (pascal(3,-1), [1,0,0;1,1,0;1,2,1])
 %!assert (pascal(3,0), [1,1,1;1,2,3;1,3,6])
 %!assert (pascal(3,0), pascal(3))
 %!assert (pascal(3,1), [1,0,0;1,-1,0;1,-2,1])
-%!assert (pascal(3,2), [0,0,-1;0,-1,2;-1,-1,1])
+%!assert (pascal(3,2), [1,1,1;-2,-1,0;1,0,0])
 %!error (pascal(3,4))
 %!error (pascal(3,-2))
 %!error (pascal())
 %!error (pascal(1,2,3))
